ostd/src/logger.rs

```diff
@@ -10,96 +10,78 @@
+        if let Some(logger) = self.backend.get() {
+            return logger.log(record);
+        };
+
+        // Default implementation.
         let level = record.level();
 
         // Use a global lock to prevent interleaving of log messages.
         use crate::sync::SpinLock;
         static RECORD_LOCK: SpinLock<()> = SpinLock::new(());
         let _lock = RECORD_LOCK.disable_irq().lock();
 
-        cfg_if::cfg_if! {
-            if #[cfg(feature = "log_color")]{
-                use owo_colors::Style;
-
-                let timestamp_style = Style::new().green();
-                let record_style = Style::new().default_color();
-                let level_style = match record.level() {
-                    log::Level::Error => Style::new().red(),
-                    log::Level::Warn => Style::new().bright_yellow(),
-                    log::Level::Info => Style::new().blue(),
-                    log::Level::Debug => Style::new().bright_green(),
-                    log::Level::Trace => Style::new().bright_black(),
-                };
-
-                crate::console::early_print(
-                    format_args!("{} {:<5}: {}\n",
-                    timestamp_style.style(format_args!("[{:>10.3}]", timestamp)),
-                    level_style.style(level),
-                    record_style.style(record.args()))
-                );
-            }else{
-                crate::console::early_print(
-                    format_args!("{} {:<5}: {}\n",
-                    format_args!("[{:>10.3}]", timestamp),
-                    level,
-                    record.args())
-                );
-            }
-        }
+        crate::console::early_print(format_args!("{}: {}\n", level, record.args()));
     }
 
-    fn flush(&self) {}
+    fn flush(&self) {
+        if let Some(logger) = self.backend.get() {
+            logger.flush();
+        };
+    }
 }
 
 /// Initialize the logger. Users should avoid using the log macros before this function is called.
 pub(crate) fn init() {
     let level = get_log_level().unwrap_or(LevelFilter::Off);
-
     log::set_max_level(level);
     log::set_logger(&LOGGER).unwrap();
 }
 
 fn get_log_level() -> Option<LevelFilter> {
-    let module_args = kernel_cmdline().get_module_args("ostd")?;
-
-    let arg = module_args.iter().find(|arg| match arg {
-        ModuleArg::Arg(_) => false,
-        ModuleArg::KeyVal(name, _) => name.as_bytes() == "log_level".as_bytes(),
-    })?;
-
-    let ModuleArg::KeyVal(_, value) = arg else {
-        unreachable!()
-    };
-    let value = value.as_c_str().to_str().unwrap_or("off");
-    Some(match value {
-        "error" => LevelFilter::Error,
-        "warn" => LevelFilter::Warn,
-        "info" => LevelFilter::Info,
-        "debug" => LevelFilter::Debug,
-        "trace" => LevelFilter::Trace,
-        // Otherwise, OFF
-        _ => LevelFilter::Off,
-    })
+    let log_level = option_env!("LOG_LEVEL")?;
```

Comment:
This PR actually changes run-time filtering to compile-time filtering. This may raise a few more questions:
 1. The kernel command line in the Makefile isn't updated properly:
```
CARGO_OSDK_ARGS := --target-arch=$(ARCH) --kcmd-args="ostd.log_level=$(LOG_LEVEL)"
```
 2. If we really want compile-time filtering, we'd better control the logging with the [features](https://docs.rs/log/latest/log/#compile-time-filters) of the `log` crate.
 3. If we want run-time filtering, how can we do it? As mentioned in https://github.com/asterinas/asterinas/pull/1502#discussion_r1818288764 above, we cannot call `log::set_max_level` before the OSTD initialization process. So this PR literally makes the logging level uncontrollable during the initialization process. Well, I'm not sure if this is a good thing.

Reply:
> This PR actually changes run-time filtering to compile-time filtering. 

I feel that this PR only converts the process of obtaining the `LOG_LEVEL` parameter into compile time. 

> we'd better control the logging with the [features](https://docs.rs/log/latest/log/#compile-time-filters) of the log crate.

The reason for not using a feature here is that using `set_max_level` is more flexible. One can  call `set_max_level` again during the initialization of the injected logger to relax this filter (like #1487 ). 

> So this PR literally makes the logging level uncontrollable during the initialization process.

I don't quite understand what you mean here. I feel that the changes in this part do not affect the setting of `log_level` during the initialization phase.

Reply:
> I feel that this PR only converts the process of obtaining the LOG_LEVEL parameter into compile time.

Yes. So we cannot change the logging level without recompiling the kernel, right? Before this PR, we could compile the kernel once and run it multiple times with different logging levels.

> The reason for not using a feature here is that using set_max_level is more flexible. One can call set_max_level again during the initialization of the injected logger to relax this filter (like https://github.com/asterinas/asterinas/pull/1487 ).
>
> I don't quite understand what you mean here. I feel that the changes in this part do not affect the setting of log_level during the initialization phase.

The Asterinas kernel cannot set the logging level of the OSTD initialization phase, right? This is actually what you said in https://github.com/asterinas/asterinas/pull/1502#discussion_r1818288764. Otherwise we can just call `log::set_logger` directly in the Asterinas kernel.

I really have no idea how to handle the logging of the OSTD initialization phase, and we can just leave that for future work if you don't have an idea either. But the fact is that we used to support runtime filtering of OSTD initialization logs, we now only support compile-time filtering, and _more importantly it is impossible to restore support in the kernel_.